emanate/__init__.py

```diff
@@ -3,77 +3,23 @@
 # Example usage:
 # TODO
 
-from argparse import ArgumentParser
+from . import config
+from argparse import ArgumentParser, SUPPRESS
 from fnmatch import fnmatch
-import json
 from pathlib import Path
 import sys
 
 class Emanate:
-    DEFAULT_IGNORE = [
-            "*~",
-            ".*~",
-            ".*.sw?",
-            "*/emanate.json",
-            "*.emanate",
-            ".*.emanate",
-            "*/.git",
-            "*/.git/*",
-            "*/.gitignore",
-            "*/.gitmodules",
-            "*/__pycache__",
-            "*/__pycache__/*",
-            ]
-
-    def __init__(self, argv):
-        """Emanate prioritizes configuration sources in the following order:
-        - default values have lowest priority;
-        - the configuration file overrides defaults;
-        - command-line arguments override everything."""
-        args   = Emanate.parse_args(argv)
-        config = Emanate.load_config(args.config)
-        self.dest   = Path(args.destination).expanduser().resolve()
-        self.ignore = self.DEFAULT_IGNORE + config.get("ignore", [])
-        self.no_confirm = args.no_confirm
-
-        if args.clean:
-            self.function = self.del_symlink
-        else:
-            self.function = self.add_symlink
-
-    @staticmethod
-    def parse_args(argv):
-        argparser = ArgumentParser(
-                description="symlink files from one directory to another")
-        argparser.add_argument("--clean",
-                action="store_true",
-                default=False,
-                help="Remove symlinks.")
-        argparser.add_argument("--destination",
-                default=Path.home(),
-                metavar="DESTINATION",
-                help="Directory to create and/or remove symlinks in.")
-        argparser.add_argument("--no-confirm",
-                action="store_true",
-                default=False,
-                help="Don't prompt before replacing a file.")
-        argparser.add_argument("--config",
-                metavar="CONFIG_FILE",
-                default="emanate.json",
-                help="Configuration file to use.")
-        return argparser.parse_args(argv[1:])
-
-    @staticmethod
-    def load_config(filename):
-        config_file = Path(filename)
-        if config_file.exists():
-            return json.loads(config_file.read_text())
-        else:
-            return {}
+    def __init__(self, *configs):
+        self.config   = config.merge(config.DEFAULTS, *configs)
+        self.dest     = self.config.destination.resolve()
+        self.function = self.del_symlink if self.config.clean else self.add_symlink
+
+        assert(config.is_resolved(self.config))
```

Comment:
The outermost parenthesis should be removed, since `assert` is a statement, not a function. (You got this right elsewhere in the code.)